### PR TITLE
Add variable option syncing for examples

### PR DESCRIPTION
**Pull request by byrnHDF**
Everything builds standalone - java examples is disabled for building during library process.

### CODE REVIEW

File: HDF5Examples/config/cmake/HDFMacros.cmake [hunk 1]
```diff
@@ -156,39 +203,10 @@ macro (HDF_DIR_PATHS package_prefix)
   endif ()
 
   SET_HDF_BUILD_TYPE()
-
 #-----------------------------------------------------------------------------
 # Setup output Directories
```
Comment (lrknox) [inline HDF5Examples/config/cmake/HDFMacros.cmake]: This comment and most of the  code it describes were copied/moved to line 13 and following.  SET_HDF_OUTPUT_DIRS is pretty much self-described.  Can this comment be removed?

Comment (byrnHDF) [reply HDF5Examples/config/cmake/HDFMacros.cmake]: Sure.

**Review by lrknox — APPROVED** [review]

**Comment by byrnHDF** [comment]
Taking WIP off as remaining issues will be addressed in followup prs.

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -47,6 +47,20 @@ New Features
...
+      As such it can build and test the examples during library build or after
+      the library is installed. Previously, the hdf5-repository archives were
+      downloaded for packaging with the library. Now the examples can be built
+      and tested without a packaged install of the library.
+
+      However to maintain the ability to use the HDF5Examples with an installed
+      library, it is necessary to translate or synch the option names from those
+      used by the library to those used by the examples. The typical pattern is:
+            <example option> = <library option>
+            HDF_BUILD_FORTRAN = ${HDF5_BUILD_FORTRAN}
```
Comment (jhendersonHDF) [inline release_docs/RELEASE.txt:62]: Will renaming these to something like HDF5EX_BUILD_XXX be done in another PR?

File: config/cmake/HDFMacros.cmake
Comment (jhendersonHDF) [inline config/cmake/HDFMacros.cmake:1]: Seems like having these duplicated CMake files under config/cmake and HDF5Examples/config/cmake is going to continue to be a maintenance burden

File: release_docs/RELEASE.txt (hunk 2 above)
Comment (byrnHDF) [reply release_docs/RELEASE.txt:62]: Yes. Need to consider all the dependencies.

File: config/cmake/HDFMacros.cmake
Comment (byrnHDF) [reply config/cmake/HDFMacros.cmake:1]: Yes, but they are needed standalone - future may come up with some kind of process, but this way someone can just copy the HDF5Examples folder and have a standalone project that can be used with a hdf5 binary.

**Review by jhendersonHDF — APPROVED** [review]
